### DISCUSSION TITLE
How to type a module that returns a string when required

### DISCUSSION THREAD

**Discussion by catwell** [teal-language/tl#628] [Q&A] [answered]
I am writing .d,tl files for an existing project (not mine). That project has a module which returns its version as a string when required, like this:

```lua
require "foo.version" -- returns "v1.2.3"
```

Is there any way to type this in Teal?

I tried this:

```
local type version = string
return version
```

but then if I do this:

```
local v: string = require("foo.version")
```

I get:

```
in local declaration: v: got type string, expected string
```

**Comment by icy-lava** [comment, marked as answer]
Late reply but - you're returning a type when you should be returning a value. So instead you can do something like `return "anything"` or `return nil as string`.

I don't know if there's a more standard way of doing this.

Could also do:
```lua
local version: string
return version
```

**Reply by hishamhm** [reply]
That `local version: string; return version` solution looks elegant for a .d.tl file!

**Reply by catwell** [reply]
Yes, I'll use that one, thanks!
